algorithms/sorting.md

```diff
@@ -8,6 +8,15 @@
 - [Recursion](/recursion/recursion.md)
 - [Searching](/algorithms/searching.md) (Particularly Binary Search)
 
+### Motivation
+
+Sorting is important in programming for the same reason it is important in everyday life. It is easier and faster to locate items in a sorted list than unsorted. Sorting algorithms can be used in a program to sort an array for later searching or writing out to an ordered file or report.
+The point isn't to remember the sorting algorithms so you can write them on the job. We of course have library algorithms/search engines for that as you said. They are taught because they serve as a good case studies on algorithm analysis/design. Of course this doesn't mean its the best way (but theres already a lot of material out there on it for teaching purposes so it's probably easier for professors to teach it too).
```

Comment:
@yati1998 please rephrase this para, try to make it easier and concise to understand.
For example, start like this:
The point here isn't to remember the sorting algorithms which you can write on your job or in an interview, but, to understand what an algorithm actually is.